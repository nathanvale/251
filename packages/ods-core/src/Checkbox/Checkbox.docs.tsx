import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { getSwitchDocs } from "../_private/components/AbstractSwitch/switch-docs";
import { Checkbox, CheckboxProps } from "./Checkbox";

export const docs: ComponentDocs<CheckboxProps> = {
  category: "Interaction",
  componentName: "Checkbox",
  description:
    "This is an accessible Checkbox component. It is uncontrolled by default. But if the `checked` value is provided it becomes controlled",
  propDescriptions: getSwitchDocs("checkbox"),
  migrationGuide: false,
  examples: {
    default: {
      Code: () => <Checkbox label="Accept Ts&Cs" id="tc-checkbox" />,
    },
    additional: [
      {
        label: "Controlled",
        description:
          "Checkbox becomes controlled when checked prop is provided.  The checked prop must NOT be undefined when used as a controlled component",
        Code: () => <Checkbox label="Accept Ts&Cs" id="tc-checkbox" checked />,
      },
      {
        label: "Error",
        description: "Only the icon turns red when it is in error state",
        Code: () => <Checkbox label="Accept Ts&Cs" id="tc-checkbox" error />,
      },
      {
        label: "Disabled",
        description: "All turn grey when it is disabled",
        Code: () => (
          <Checkbox label="Accept Ts&Cs" id="tc-checkbox" disabled checked />
        ),
      },
      {
        label: "Showing Helper Text",
        description:
          "One could put a helper text under the label for further details",
        Code: () => (
          <Checkbox
            label="Accept Ts&Cs"
            id="tc-checkbox"
            helperText="Make sure you read our Ts&Cs carefully."
          />
        ),
      },
    ],
  },
  snippets: [
    {
      label: "Uncontrolled",
      Code: () => <Checkbox label="Checkbox 1" id="checkbox1" />,
    },
  ],
};
