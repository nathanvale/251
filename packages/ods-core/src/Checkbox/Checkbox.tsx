import React from "react";
import { CheckboxProps as MuiCheckboxProps } from "@material-ui/core";
import { BaseSwitchProps } from "@origin-digital/ods-types";
import { CheckboxBase } from "../CheckboxBase/CheckboxBase";
import { AbstractSwitch } from "../_private/components/AbstractSwitch/AbstractSwitch";
import { SwitchMuiProps } from "../_private/components/AbstractSwitch/abstract-types";

export type CheckboxMuiPropsType = SwitchMuiProps<MuiCheckboxProps>;

export interface CheckboxProps extends BaseSwitchProps {
  muiProps?: CheckboxMuiPropsType;
}

export const Checkbox = (props: CheckboxProps) => {
  return (
    <AbstractSwitch<MuiCheckboxProps, CheckboxMuiPropsType>
      Component={CheckboxBase}
      {...props}
    />
  );
};

Checkbox.displayName = "Checkbox";
Checkbox.defaultProps = {};
