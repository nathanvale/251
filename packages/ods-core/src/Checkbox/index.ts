export { Checkbox } from "./Checkbox";
export type CheckboxProps = import("./Checkbox").CheckboxProps;
