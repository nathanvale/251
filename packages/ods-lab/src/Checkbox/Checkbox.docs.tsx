/* eslint-disable react/display-name */
import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { ContentSection, Checkbox, CheckboxProps } from "..";

export const docs: ComponentDocs<CheckboxProps> = {
  category: "Experimental",
  componentName: "Checkbox",
  description: "",
  propDescriptions: {},
  migrationGuide: false,
  examples: [
    {
      Code: () => (
        <ContentSection backgroundColor="white">
          <Checkbox
            name="checkbox"
            label="checkbox"
            helperText="helperText"
            onChange={() => {}}
            checked={true}
            error={false}
          />
        </ContentSection>
      ),
    },
  ],
  snippets: [],
};
