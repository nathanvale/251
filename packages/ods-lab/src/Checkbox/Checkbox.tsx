import React from "react";
import MuiCheckbox, {
  CheckboxProps as MUICheckboxProps,
} from "@material-ui/core/Checkbox";
import {FormControlLabel} from "../_private/components/FormControlLabel/FormControlLabel";
import {FormHelperText} from "../_private/components/FormHelperText/FormHelperText";

export interface CheckboxProps {
  name: string;
  checked: boolean;
  "data-id"?: string;
  label: string;
  error?: boolean;
  onChange: MUICheckboxProps["onChange"];
  required?: boolean;
  disabled?: boolean;
  className?: string;
  helperText?: string;
  value?: string | number | string[];
}

export const Checkbox = (props: CheckboxProps) => {
  const {
    "data-id": dataId,
    disabled,
    label,
    error,
    helperText,
    ...other
  } = props;
  return (
    <>
      <FormControlLabel
        data-id={dataId}
        disabled={disabled}
        label={label}
        control={
          <MuiCheckbox
            inputProps={{"aria-label": label}}
            disabled={disabled}
            {...other}
          />
        }
      />
      <FormHelperText>{helperText}</FormHelperText>
      {error && <FormHelperText error={error}>{helperText}</FormHelperText>}
    </>
  );
};
