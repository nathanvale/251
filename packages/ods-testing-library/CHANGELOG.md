# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
