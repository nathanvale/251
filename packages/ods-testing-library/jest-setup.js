import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
