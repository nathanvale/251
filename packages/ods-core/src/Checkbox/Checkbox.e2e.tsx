/* eslint-disable react/display-name */
import * as React from "react";
import { E2ETests } from "@origin-digital/ods-types";
import { Checkbox } from "./Checkbox";

export const tests: E2ETests = [
  {
    label: "Unchecked Checkbox",
    Code: () => <Checkbox id="checkbox" label="Unchecked" />,
  },
  {
    label: "Unchecked disabled Checkbox",
    Code: () => <Checkbox id="checkbox" label="Unchecked" disabled />,
  },
  {
    label: "Unchecked error Checkbox",
    Code: () => <Checkbox id="checkbox" label="Unchecked" error />,
  },
  {
    label: "Checked Checkbox",
    Code: () => <Checkbox id="checkbox" label="Checked" checked />,
  },
  {
    label: "Checked disabled Checkbox",
    Code: () => <Checkbox id="checkbox" label="Checked" checked disabled />,
  },
  {
    label: "Checked error Checkbox",
    Code: () => <Checkbox id="checkbox" label="Checked" checked error />,
  },
  {
    label: "Checkbox wit helperText",
    Code: () => (
      <Checkbox id="checkbox" label="chk-helper" helperText="More info" />
    ),
  },
];
