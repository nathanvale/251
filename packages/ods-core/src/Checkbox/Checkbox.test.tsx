import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { Checkbox } from "./Checkbox";

test("it renders an unchecked Checkbox with label and helperText", () => {
  const { container } = render(
    <Checkbox label="Check 1" id="my-checkbox" helperText="This is more info" />
  );

  expect(container).toMatchSnapshot();
});

test("it shows correct label", () => {
  const { getAllByText } = render(
    <Checkbox label="Check 1" id="my-checkbox" />
  );

  const items = getAllByText("Check 1");
  expect(items).toHaveLength(1);
  expect(items[0]).toHaveClass("MuiFormControlLabel-label");
});

test("it renders a Checkbox in error state correctly", () => {
  const { container } = render(
    <Checkbox label="Check 1" id="my-checkbox" error />
  );

  expect(container).toMatchSnapshot();
});

test("it renders a disabled Checkbox", () => {
  const { container } = render(
    <Checkbox label="Check 1" id="my-checkbox" disabled />
  );

  expect(container).toMatchSnapshot();
});
