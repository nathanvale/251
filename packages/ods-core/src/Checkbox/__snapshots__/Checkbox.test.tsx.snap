// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a Checkbox in error state correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root AbstractSwitch-formControlLabel-173"
      data-id="checkbox"
      id="my-checkbox-label"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-190 MuiCheckbox-root AbstractSwitchBase-error-root-182 AbstractSwitchBase-error-root-183 MuiCheckbox-colorSecondary AbstractSwitch-switchComp-174 MuiIconButton-colorSecondary"
        data-id="checkbox-CheckboxBase"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-checkbox-helperText"
            class="PrivateSwitchBase-input-193"
            data-indeterminate="false"
            id="my-checkbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m2-2H3v18h18V3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
  </div>
</div>
`;

exports[`it renders a disabled Checkbox 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root AbstractSwitch-formControlLabel-255 Mui-disabled"
      data-id="checkbox"
      id="my-checkbox-label"
    >
      <span
        aria-disabled="true"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-272 MuiCheckbox-root AbstractSwitchBase-root-264 AbstractSwitchBase-root-265 MuiCheckbox-colorSecondary AbstractSwitch-switchComp-256 PrivateSwitchBase-disabled-274 Mui-disabled MuiIconButton-colorSecondary Mui-disabled Mui-disabled"
        data-id="checkbox-CheckboxBase"
        tabindex="-1"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-checkbox-helperText"
            class="PrivateSwitchBase-input-275"
            data-indeterminate="false"
            disabled=""
            id="my-checkbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m2-2H3v18h18V3z"
            />
          </svg>
        </span>
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label Mui-disabled MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
  </div>
</div>
`;

exports[`it renders an unchecked Checkbox with label and helperText 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

<div>
  <div
    class="c0"
  >
    <label
      class="MuiFormControlLabel-root AbstractSwitch-formControlLabel-1"
      data-id="checkbox"
      id="my-checkbox-label"
    >
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-18 MuiCheckbox-root AbstractSwitchBase-root-10 AbstractSwitchBase-root-11 MuiCheckbox-colorSecondary AbstractSwitch-switchComp-2 MuiIconButton-colorSecondary"
        data-id="checkbox-CheckboxBase"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            aria-describedby="my-checkbox-helperText"
            class="PrivateSwitchBase-input-21"
            data-indeterminate="false"
            id="my-checkbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m2-2H3v18h18V3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Check 1
      </span>
    </label>
    <p
      class="MuiFormHelperText-root AbstractSwitch-formHelperText-3"
      data-id="checkbox-helperText"
      id="my-checkbox-helperText"
    >
      This is more info
    </p>
  </div>
</div>
`;
